sycl/doc/UsersManual.md

```diff
@@ -300,6 +301,16 @@ and not recommended to use in production environment.
     The forced target applies to all objects, archives and default device
     libraries.
 
+**`-f[no-]sycl-rdc`**
+
+    Enables/disables relocatable device code. If relocatable device code is
+    disabled, device code cannot use SYCL_EXTERNAL functions, which allows
+    the compiler to link device code on a per-translation-unit basis.
+    This may result in compile time and compiler memory usage improvements.
+    '-fno-sycl-rdc' used along with '-fsycl-max-parallel-link-jobs' will enable
+    additional device linking parallism for fat static archives.
+    Relocatable device code is enabled by default.
```

Comment:
Why do we call this mode "relocatable"?  I don't see how this option makes the device code any more (or less) relocatable.

I think a better name for the option would be something like **-fsycl-link-device-code-at-compile-time** or maybe **-fsycl-link-device-code=\<mode\>**, where **\<mode\>** could be either **compile-time** or **link-time**.

Reply:
Hey Greg,
We're following the community naming convention for this. 

https://reviews.llvm.org/D52377

"When -fgpu-rdc is off,
clang will assume the device code in each translation unit does not call
external functions except those in the device library, therefore it is possible
to compile the device code in each translation unit to self-contained kernels
and embed them in the host object, so that the host object behaves like
usual host object which can be linked by lld."

Reply:
I still don't understand why the word "relocatable" makes sense, but I guess it's good to follow an established convention.  Is the idea that "relocatable" means that the kernel code can call external functions?  That's not the usual meaning of that word from my experience.

I did notice in the link you provided that they list these advantages when disabling "relocatable" device code:

1. allow users to create static libraries which can be linked by host linker
2. amortized device code linking time

Our description lists (1) as an advantage, but it doesn't say anything about (2).  Should it?

Reply:
My understanding of the usage of relocatable here matches yours about calling external functions. 

Sorry, do you have the numbers swapped in your last line? We do list device code linking time improvements here:
"This may result in compile time and compiler memory usage improvements."

Reply:
Sorry, yes, my numbers were switched.  We do not list (1) as an advantage.  Does this apply to DPC++?